msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase2582290.java

```diff
@@ -0,0 +1,160 @@
+
+        // Install new Authenticator with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install old MSALTestApp
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.installOldApk();
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+
+        // Click on "AcquireToken" button
+        // User is Prompted for creds
+        final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                .prompt(PromptParameter.SELECT_ACCOUNT)
+                .loginHint(username)
+                .sessionExpected(false)
+                .broker(mBroker)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingProvidedAccountInBroker(false)
+                .expectingLoginPageAccountPicker(false)
+                .expectingProvidedAccountInCookie(false)
+                .consentPageExpected(false)
+                .passwordPageExpected(true)
+                .speedBumpExpected(false)
+                .registerPageExpected(false)
+                .enrollPageExpected(false)
+                .staySignedInPageExpected(false)
+                .verifyYourIdentityPageExpected(false)
+                .howWouldYouLikeToSignInExpected(false)
+                .build();
+
+        // Enter username and pwd from step 2 to complete the acquire token flow
+        // Token should be retrieved succeffully
+        final String token = msalTestApp.acquireToken(username, password, promptHandlerParameters, true);
+        Assert.assertNotNull(token);
+        msalTestApp.handleBackButton();
+
+        // Click on "Get Users" button
+        // The user account should be shown in the UI
+        final List<String> users = msalTestApp.getUsers();
+        Assert.assertTrue(users.size() == 1);
+        Assert.assertTrue(users.get(0).contains(username));
+        msalTestApp.handleBackButton();
+
+        // Click on "Acquire Token Silent" button
+        // Token should be retrieved successfully
+        final String silentToken = msalTestApp.acquireTokenSilent();
+        Assert.assertNotNull(silentToken);
+        msalTestApp.handleBackButton();
+
+        // Select the Auth scheme as "POP"
+        msalTestApp.selectFromAuthScheme("POP");
+
+        // Click on "Generate SHR" button
+        // UI should be updated with an SHR token
+        final String shrToken = msalTestApp.generateSHR();
+        Assert.assertNotNull(shrToken);
+        msalTestApp.handleBackButton();
+
+        // Click on "Remove User" button
+        // UI updated with message "The account is successfully removed"
+        final String removeUserMessage = msalTestApp.removeUser();
+        Assert.assertEquals("The account is successfully removed.", removeUserMessage);
```

Comment:
can we move this to inside of removeUser method?

Reply:
Yes, we can, but it would require another round of infra change. If it's not necessary, I suggest we can do some refactor for it later since this change won't affect the result of the test case.